### ISSUE TITLE
Add `--ci` mode tuned for CI usage

### ISSUE THREAD

**Issue by Daenyth** [google/yapf#306]
- Don't report diff or modify, instead print a list of files that would be reformatted
- Return a non-zero exit code if changes are needed (could be a count of files needing changes)

**Comment by travcunn** [comment]
You could run this to count how many files would be reformatted:

```
yapf --diff --recursive . | grep "     (reformatted)" | wc -l
```

**Comment by Daenyth** [comment]
The problem there is that recursive mode is slow. It takes about 10 minutes
on my codebase. Right now I have a script that uses git to list each
changed file and launches separate yapf processes for each file then
collects the error codes. That usually only takes some seconds

**Comment by gwelymernans** [comment]
The way that you describe `--ci` it seems very similar to a `--dry-run` flag. It makes me wonder about the use case you want this flag for. Could you describe in which situations it would be used and how?

**Comment by Socialery** [comment]
most continuous integration methods use the return code of the programs it runs, to tell if there has been a fail with the tests

a hypothetical .travis.yml

```yml

language: python
python:
  - "3.5"
script:
  - python -m unittest discover -v
  - yapf --recursive  # --ci
```
unittest works fine, it signals that there was an "error" with the tests, by returning a non-zero when it `sys.exit`s

yapf on the other hand currently doesn't signal to the outside world its results, other than via stdout

Why would someone want this via continuous integration though? because CI can also be ran on pull requests! so while it doesn't help so much for committing straight to master, it can help keep a code-base clean when working collaboratively.

So the most important thing for a `--ci` would have to be the return code, I'd imagine. Sorry if this response is a bit wordy!

**Comment by Daenyth** [comment]
I have yapf configured to run on my CI server so that usage is enforced. I have that set up to prevent large diffs from accumulating over time as some developers use yapf, some use IDE reformat, some manually format, etc.
